cvat/apps/engine/cloud_provider.py

```diff
@@ -248,16 +247,35 @@ def list_files_on_one_page(
         _use_flat_listing: bool = False,
         _use_sort: bool = False,
     ) -> Dict:
-        result = self._list_raw_content_on_one_page(prefix, next_token, page_size)
+        search_prefix = prefix
+        if self.prefix and (len(prefix or "") < len(self.prefix)):
+            next_layer_and_tail = self.prefix[(prefix or "").find('/') + 1:].split(
+                "/", maxsplit=1
+            )
+            if 2 == len(next_layer_and_tail):
+                directory = (
+                    next_layer_and_tail[0]
+                    if not _use_flat_listing
+                    else self.prefix[: (prefix or "").find('/') + 1] + next_layer_and_tail[0] + "/"
+                )
+                return {
+                    "content": [{"name": directory, "type": "DIR"}],
+                    "next": None,
+                }
+            else:
+                search_prefix = self.prefix
```

Comment:
Let's assume that `self.prefix = "a/b/c/d_"` and `prefix = "e/b"` then this code will return `b` (or `a/b`) is that expected behaviour? Because to me it looks strange and I expect to see an empty list. Or am I missing something?